docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+ENCRYPTION_PROVIDER_CONFIG=$(ps -ef | grep kube-apiserver | grep -- --encryption-provider-config | sed 's%.*encryption-provider-config[= ]\([^ ]*\).*%\1%')
+if test -e $ENCRYPTION_PROVIDER_CONFIG; then grep -A1 'providers:' $ENCRYPTION_PROVIDER_CONFIG | tail -n1 | grep -o "[A-Za-z]*" | sed 's/^/provider=/'; fi
+```
+
+**Expected output:**
+
+```
+--encryption-provider-config is one of or all of
+aescbc,kms,secretbox
+```
+
+##### Control 1.2.31
+
+**Description:**
+
+Ensure that the API Server only makes use of Strong
+Cryptographic Ciphers (Manual)
+
+
+**Remediation:**
+
+Edit the API server configuration file
+/etc/kubernetes/manifests/kube-apiserver.yaml
+on the control plane node and set the following argument.
+
+```
+--tls-cipher-suites=TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_S
+HA384,TLS_CHACHA20_POLY1305_SHA256,
+TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AE
+S_128_GCM_SHA256,
+TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AE
+S_256_GCM_SHA384,
+TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_
+CHACHA20_POLY1305_SHA256,
+TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_1
+28_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_25
+6_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
+TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,TLS_RSA_WITH_3DE
+S_EDE_CBC_SHA,TLS_RSA_WITH_AES_128_CBC_SHA,
+TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_CBC_SHA
+,TLS_RSA_WITH_AES_256_GCM_SHA384
+```
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--tls-cipher-suites=TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA38
+4,TLS_CHACHA20_POLY1305_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_
+SHA,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH
+_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECD
+HE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_PO
+LY1305_SHA256,TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,TLS_ECDHE_RSA_
+WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_E
+CDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA
+384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_CHAC
+HA20_POLY1305_SHA256,TLS_RSA_WITH_3DES_EDE_CBC_SHA,TLS_RSA_WITH_
+AES_128_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES
+_256_CBC_SHA,TLS_RSA_WITH_AES_256_GCM_SHA384
+```
+
+#### Controller Manager
+
+##### Control 1.3.1
+
+**Description:**
+
+Ensure that the --terminated-pod-gc-threshold argument is
+set as appropriate (Manual)
+
+
+**Remediation:**
+
+Edit the Controller Manager configuration file
+/var/snap/k8s/common/args/kube-controller-manager
+on the control plane node and set the --terminated-pod-gc-
+threshold to an appropriate threshold.
+
+`--terminated-pod-gc-threshold=10`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-controller-manager | grep -v grep
+```
+
+**Expected output:**
+
+```
+--terminated-pod-gc-threshold=12500
```

Comment:
threshold is set to 10 in the remediation but 12500 in the expected output. Does this matter?

Reply:
I aligned the kube-bench tests to recommend the 12500 value.